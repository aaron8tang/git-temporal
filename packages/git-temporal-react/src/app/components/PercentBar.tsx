import React from 'react';
import { style } from 'app/styles';

export interface PercentBarProps {
  numerator: number;
  denominator: number;
}

const defaultNumeratorBarStyle = {
  backgroundColor: 'blue',
  height: 10,
};
const defaultDenominatorBarStyle = {
  backgroundColor: 'grey',
  height: 10,
  marginBottom: 5,
};

// numbers for humans have commas
export const PercentBar = ({
  numerator,
  denominator,
}: PercentBarProps): JSX.Element => {
  const percentWidth = (numerator / denominator) * 100;
  return (
    <div style={style(defaultDenominatorBarStyle, { width: '100%' })}>
      <div
        style={style(defaultNumeratorBarStyle, { width: `${percentWidth}%` })}
      />
    </div>
  );
};

PercentBar.displayName = 'PercentBar';
