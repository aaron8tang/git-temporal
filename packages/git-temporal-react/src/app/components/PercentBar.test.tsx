import React from 'react';
import { shallow } from 'enzyme';

import { PercentBar } from './PercentBar';

describe('components/PercentBar', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<PercentBar numerator={70} denominator={100} />);
    });

    test('it should match snapshot (it should have rendered a 70% width div within a 100% width div)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
