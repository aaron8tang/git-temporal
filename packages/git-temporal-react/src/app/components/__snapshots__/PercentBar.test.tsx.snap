// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PercentBar when rendered with required props it should match snapshot (it should have rendered a 70% width div within a 100% width div) 1`] = `
<div
  style={
    Object {
      "backgroundColor": "grey",
      "height": 10,
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "height": 10,
        "width": "70%",
      }
    }
  />
</div>
`;
